hooks/useInternalRouter.ts

```
@@ -7,29 +7,58 @@ export const useInternalRouter = () => {
       }
-      const searchParams = new URLSearchParams();
-      // eslint-disable-next-line no-restricted-syntax
-      for (const [key, value] of Object.entries(query)) {
-        if (value !== undefined) {
-          searchParams.set(key, String(value));
+    ) => {
+      const { params, query } = options ?? {};
+
+      let url = basePath;
+      if (params && params.length > 0) {
+        url += '/' + params.map(String).join('/');
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)